src/examples/table/TableSort.tsx

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useState } from 'react';
+import {
+  Body,
+  Cell,
+  Head,
+  HeaderCell,
+  HeaderRow,
+  Row,
+  SortableCell,
+  Table
+} from '@zendeskgarden/react-tables';
+
+interface IRowData {
+  id: string;
+  subject: string;
+  requester: string;
+  type: string;
+}
+
+type Direction = 'asc' | 'desc' | undefined;
+
+const rowData: IRowData[] = [];
+
+for (let x = 0; x < 10; x++) {
+  rowData.push({
+    id: `row-${x}`,
+    subject: `Custom ticket view ${x + 1}`,
+    requester: x % 2 === 0 ? 'John Smith' : 'Jane Doe',
+    type: x % 3 === 0 ? 'Ticket' : 'Incident'
+  });
+}
+
+const sortData = (tableData: IRowData[], requesterSort: Direction, typeSort: Direction) => {
+  if (!requesterSort && !typeSort) {
+    return tableData;
+  }
+
+  let field: string;
+  let sortValue: Direction;
+
+  if (requesterSort) {
+    field = 'requester';
+    sortValue = requesterSort;
+  } else {
+    field = 'type';
+    sortValue = typeSort;
+  }
+
+  return tableData.sort((a: any, b: any) => {
+    const aValue = a[field];
+    const bValue = b[field];
+
+    if (aValue > bValue) {
+      return sortValue === 'asc' ? 1 : -1;
+    } else if (aValue < bValue) {
+      return sortValue === 'asc' ? -1 : 1;
+    }
+
+    return 0;
+  });
+};
+
+const Example = () => {
+  const [data, setData] = useState(rowData);
+  const [requesterSort, setRequesterSort] = useState<Direction>();
+  const [typeSort, setTypeSort] = useState<Direction>();
+
+  return (
+    <div style={{ minWidth: 500 }}>
+      <Table>
+        <Head>
+          <HeaderRow>
+            <HeaderCell>Subject</HeaderCell>
+            <SortableCell
+              onClick={() => {
+                if (requesterSort === 'asc') {
+                  setData(data);
+                  setRequesterSort('desc');
+                  setTypeSort(undefined);
+                } else if (requesterSort === 'desc') {
+                  setData(data);
+                  setRequesterSort(undefined);
+                  setTypeSort(undefined);
+                } else {
+                  setData(data);
+                  setRequesterSort('asc');
+                  setTypeSort(undefined);
```

Comment:
I know these were mostly copied from the existing examples, but I noticed that some of these `setStates` are called in every possible logic branch. Which means the calls can be combined into a single call without affecting the logic (I think):

```tsx
<SortableCell
  onClick={() => {
    if (requesterSort === 'asc') {
      setRequesterSort('desc');
    } else if (requesterSort === 'desc') {
      setRequesterSort(undefined);
    } else {
      setRequesterSort('asc');
    }
    setTypeSort(undefined);
    setData(data);
  }}
  sort={requesterSort}
>
  Requester
</SortableCell>
```

Reply:
@austin94 Lemme know what you think for this and if we should revert to the [single setState call](https://github.com/zendeskgarden/website/pull/172/commits/87d755f14a986714af40f9ddb6d1ff42cedbd366#diff-bc140959802676c8fdb6126b94a27e56R71). You mentioned sometimes it's a best practice to separate them out.

Reply:
Consider this:

```js
if (hungry) {
  eat();
  relax();
} else if (thirsty) {
  drink();
  relax();
} else {
  relax();
};
```

There isn't a need to call `relax` on every logical branch if you know it must happen all the time. This would be preferred:

```js
if (hungry) {
  eat();
} else if (thirsty) {
  drink();
};

relax();
```

The same can be said of these function calls:

```tsx
setTypeSort(undefined);
setData(data);
```

Reply:
Nice! Much better